paddle/framework/tensor.h

```diff
@@ -127,6 +128,10 @@ class Tensor {
 
   DDim dims() const { return dims_; }
 
+  platform::Place place() const { return holder_->place(); }
```

Comment:
This method `Tensor::place` would segment fault if a Tensor is constructed by its `mutable_data` hasn't been called.  Given such delicacy, I wouldn't expose it to all users.

It seems that `Tensor::place` is exposed only for pybind.  If so, a friend class to Python exposure might be a better design choice -- we just make sure that this friend class calls `mutable_data` before access `holder_->place_`, or doesn't necessary check and returns errors to Python.

In the future, maybe all C++ types need to be exposed to Python could have such a friend class.